### PR TITLE
Deprecating unused classes

### PR DESCRIPTION
**Pull request by escopecz**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [Y]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes https://github.com/mautic/mautic/pull/13064

#### Description:

@TomasVotruba [pointed out](https://github.com/mautic/mautic/pull/13064) we have some unused classes. I went through them and created this list of classes we can deprecate and remove in M6

#### Steps to test this PR:

1. No need to test. This PR is just adding the deprecation comments

### CODE REVIEW

File: app/bundles/MessengerBundle/DependencyInjection/EnvProcessor/MessengerNullableEnvVarProcessor.php [hunk 1]
```diff
@@ -6,6 +6,9 @@
 
 use Symfony\Component\DependencyInjection\EnvVarProcessorInterface;
 
+/**
+ * @deprecated since Mautic 5.0, to be removed in 6.0 with no replacement.
+ */
```
Comment (escopecz) [inline app/bundles/MessengerBundle/DependencyInjection/EnvProcessor/MessengerNullableEnvVarProcessor.php]:
```suggestion
```
This is being used https://github.com/mautic/mautic/blob/a33b41b32a2bce629b15c6098a71da071a080a4c/app/config/config.php#L95

File: app/bundles/ApiBundle/Entity/oAuth2/RefreshToken.php [hunk 2]
```diff
@@ -6,6 +6,9 @@
 use FOS\OAuthServerBundle\Model\RefreshToken as BaseRefreshToken;
 use Mautic\CoreBundle\Doctrine\Mapping\ClassMetadataBuilder;
 
+/**
+ * @deprecated since Mautic 5.0, to be removed in 6.0 with no replacement.
+ */
```
Comment (escopecz) [inline app/bundles/ApiBundle/Entity/oAuth2/RefreshToken.php]: Needed for https://github.com/FriendsOfSymfony/FOSOAuthServerBundle/blob/master/Resources/doc/index.md#doctrine-orm-classes
```suggestion
```

File: app/bundles/LeadBundle/Form/Validator/Constraints/UniqueCustomFieldValidator.php [hunk 3]
```diff
@@ -14,6 +14,9 @@
 use Symfony\Component\Validator\Constraint;
 use Symfony\Component\Validator\ConstraintValidator;
 
+/**
+ * @deprecated since Mautic 5.0, to be removed in 6.0 with no replacement.
+ */
```
Comment (escopecz) [inline app/bundles/LeadBundle/Form/Validator/Constraints/UniqueCustomFieldValidator.php]:
```suggestion
```
Used in Company and Lead entities

File: app/bundles/EmailBundle/Validator/DsnValidator.php [hunk 4]
```diff
@@ -14,6 +14,9 @@
 use Symfony\Component\Validator\ConstraintValidator;
 use Symfony\Component\Validator\Exception\UnexpectedTypeException;
 
+/**
+ * @deprecated since Mautic 5.0, to be removed in 6.0 with no replacement.
+ */
```
Comment (escopecz) [inline app/bundles/EmailBundle/Validator/DsnValidator.php]:
```suggestion
```
Used in Email ConfigType

**Review by kuzmany — APPROVED** [review]
Let's do it 👍

**Pull request merged**